lib/main.dart

```diff
@@ -1,43 +1,55 @@
-import 'package:clnapp/api/api.dart';
-import 'package:clnapp/api/client_provider.dart';
-import 'package:clnapp/api/cln/cln_client.dart';
+import 'package:clnapp/constants/user_setting.dart';
+import 'package:clnapp/helper/settings/get_settings.dart';
 import 'package:clnapp/utils/app_provider.dart';
 import 'package:clnapp/views/app_view.dart';
 import 'package:clnapp/views/home/home_view.dart';
+import 'package:clnapp/views/setting/setting_view.dart';
 import 'package:flutter/material.dart';
 import 'package:trash_themes/themes.dart';
-import 'dart:io' show Platform;
 
 Future<void> main() async {
   var provider = await AppProvider().init();
 
   /// TODO just for now we insert by hand GRPC information, but in the future
   /// we need to have a Setting UI!
-  var certificateDir = Platform.environment['CLN_CERT_PATH'];
-  if (certificateDir == null) {
-    throw Exception("Please export the CLN_CERT_PATH for your system");
-  }
-  provider.registerLazyDependence<AppApi>(() {
-    return CLNApi(
-        mode: ClientMode.grpc,
-        client: ClientProvider.getClient(mode: ClientMode.grpc, opts: {
-          'certificatePath': certificateDir,
-          'host': 'localhost',
-          'port': 8001,
-        })
-        // mode: ClientMode.unixSocket,
-        // client: ClientProvider.getClient(mode: ClientMode.unixSocket, opts: {
-        //   // include the path if you want use the unix socket. N.B it is broken!
-        //   'path': "$certificateDir/lightning-rpc",
-        // })
-        );
-  });
-  runApp(CLNApp(provider: provider));
+  // var certificateDir = Platform.environment['CLN_CERT_PATH'];
+  // if (certificateDir == null) {
+  //   throw Exception("Please export the CLN_CERT_PATH for your system");
+  // }
+  Setting setting = Setting();
+  await getSettingsInfo().then((value) => {
+        setting = value,
+      });
+  // provider.registerLazyDependence<AppApi>(() {
+  //   if(setting.connectionType==clients[0]){
+  //     return CLNApi(
+  //         mode: ClientMode.grpc,
+  //         client: ClientProvider.getClient(mode: ClientMode.grpc, opts: {
+  //           ///FIXME: make a login page and take some path as input
+  //           'certificatePath': setting.path,
+  //           'host': setting.host,
+  //           'port': 8001,
+  //         })
+  //     );
+  //   }
+  //   else{
+  //     return CLNApi(
+  //       mode: ClientMode.unixSocket,
+  //         ///FIXME: make a login page and take some path as input
+  //       client: ClientProvider.getClient(mode: ClientMode.unixSocket, opts: {
+  //         // include the path if you want use the unix socket. N.B it is broken!
+  //         'path': "${setting.path}/lightning-rpc",
+  //       })
+  //     );
+  //   }
+  // });
```

Comment:
I think we should keep this? or maybe make it a little bit better? because if I close the app and reload it I receive the following error

![Selection_059](https://user-images.githubusercontent.com/17150045/183296051-79a029c7-3014-4dff-b754-83580a4405c4.png)